src/index.ts

```diff
@@ -163,6 +163,25 @@ function FlatpickrInstance(
     }
   }
 
+  function getHourElementValueAsMilitary24(): number {
+    const hourElementNumberValue = self.hourElement
+      ? +self.hourElement.value
+      : self.config.defaultHour;
+
+    return self.amPM !== undefined && hourElementNumberValue
+      ? ampm2military(hourElementNumberValue, self.amPM.textContent as string)
+      : hourElementNumberValue;
+  }
+
+  function isSameTimeAsDefault(): boolean {
+    return (
+      self.hourElement !== undefined &&
+      self.minuteElement !== undefined &&
+      getHourElementValueAsMilitary24() === self.config.defaultHour &&
```

Comment:
Does the defaultHour take the am/pm config in consideration ? If not, the condition will be false when it should be true

Reply:
No the config should be 24 hours. In ADS we use 0 anyway as per spec so that won't be a problem